Pipes - Amy Cash - Solar System.rb

```diff
@@ -0,0 +1,196 @@
+#Wave Three (Wave One Hash Version follows, but is commented out; Wave Two is encapsulated in Wave Three)
+
+class Planet
+
+  attr_reader :name, :namesake, :type, :year, :distance 
+
+  def initialize(name, namesake, type, year, distance)
+    @name = name
+    @namesake = namesake
+    @type = type
+    @year = year
+    @distance = distance
+  end
+
+  def give_info
+    list =  "#{@name}: \tNamed for: #{@namesake} \tType: #{type} \tYear: #{year} Earth Years \tDistance: \t#{distance} million kilometers."
+    return list
+  end
+
+end
+
+class SolarSystem
+
+  def initialize(list_of_planets)
+    @planets = list_of_planets
+  end
+
+  def add_planet(planet)
+    @planets += planet
+  end
+
+  def print_list
+    list = ""
+    @planets.each_with_index do |planet, index|
+      list += @planets[index].give_info + "\n"
+    end
+    return list
+  end
+
+  def check_input(input)
+    until input.is_a?(Integer) && input.between?(1, 9)
+      print "Please enter a number between 1 and 9: "
```

Comment:
I love the idea of putting this in a method. However, I'm not sure it has anything to do with solar systems - it might be better as a freestanding method.